tests/unit_tests/beamlines/i24/jungfrau_commissioning/test_do_manual_acquisition.py

```diff
@@ -0,0 +1,80 @@
+import asyncio
+from unittest.mock import AsyncMock, MagicMock, patch
+
+import bluesky.plan_stubs as bps
+from bluesky.preprocessors import run_decorator
+from bluesky.run_engine import RunEngine
+from bluesky.simulators import RunEngineSimulator, assert_message_and_return_remaining
+from ophyd_async.core import AutoIncrementFilenameProvider, StaticPathProvider
+from ophyd_async.fastcs.jungfrau import Jungfrau
+from ophyd_async.testing import set_mock_value
+
+from mx_bluesky.beamlines.i24.jungfrau_commissioning.do_external_acquisition import (
+    do_external_acquisition,
+)
+
+
+async def _do_sleep():
+    await asyncio.sleep(0)
+
+
+def test_full_do_external_acquisition(jungfrau: Jungfrau, RE: RunEngine, caplog):
+    @run_decorator()
+    def test_plan():
+        status = yield from do_external_acquisition(0.001, 0.002, 5, 5, jungfrau)
+        assert not status.done
+        val = 0
+        while not status.done:
+            val += 1
+            set_mock_value(jungfrau._writer._drv.num_captured, val)
+
+            # Let status update
+            yield from bps.wait_for([_do_sleep])
```

Comment:
Nit: could use a partial here
```suggestion
            yield from bps.wait_for([partial(asyncio.sleep,0)])
```